backend/src/routes.js

```diff
@@ -144,6 +149,57 @@ router.get('/ngos/:id', jwtMiddleware({ secret: process.env.ACCESS_TOKEN_KEY, al
     }
 });
 
+router.post('/ngos/:id/upload', jwtMiddleware({ secret: process.env.ACCESS_TOKEN_KEY, algorithms: ['HS256'] }), validateRoles(['user-ngo', 'user-independent', 'admin']), async(req, res) => {
+    const checkId = await User.findById(req.user.id);
+    if (!(checkId.affiliated_ngo.ID === req.params.id)) {
+        res.status(403).json({
+            error: 'Error during image/document upload.'
+        });
+    }
+    const fileInfos = req.files.file;
+    const prExt = /jpg|jpeg|png|gif|pdf/;
+    const checkExt = prExt.test(path.extname(fileInfos[0].name));
+    const checkmime = prExt.test(fileInfos[0].mimetype);
+
+    if (checkExt && checkmime) {
+        mongoose.connect(process.env.MONGODB_KEY);
+        const { connection } = mongoose;
+        grid.mongo = mongoose.mongo;
```

Comment:
Logic looks great, just make sure that this works with mongoose-gridfs. Let me know if you run into any issues.